hello-vulkan/app/src/main/cpp/vk_triangle/hellovk.h

```diff
@@ -0,0 +1,1279 @@
+  assetManager = newManager;
+  if (initialized) {
+    createSurface();
+    recreateSwapChain();
+  }
+}
+
+void HelloVK::recreateSwapChain() {
+  vkDeviceWaitIdle(device);
+  cleanupSwapChain();
+  createSwapChain();
+  createImageViews();
+  createFramebuffers();
+}
+
+void HelloVK::render() {
+  if (orientationChanged) {
+    onOrientationChange();
+  }
+
+  vkWaitForFences(device, 1, &inFlightFences[currentFrame], VK_TRUE,
+                  UINT64_MAX);
+  uint32_t imageIndex;
+  VkResult result = vkAcquireNextImageKHR(
+      device, swapChain, UINT64_MAX, imageAvailableSemaphores[currentFrame],
+      VK_NULL_HANDLE, &imageIndex);
+  if (result == VK_ERROR_OUT_OF_DATE_KHR) {
+    recreateSwapChain();
+    return;
+  } else if (result != VK_SUCCESS && result != VK_SUBOPTIMAL_KHR) {
+    throw std::runtime_error("failed to acquire swap chain image!");
+  }
+  updateUniformBuffer(currentFrame);
+
+  vkResetFences(device, 1, &inFlightFences[currentFrame]);
+  vkResetCommandBuffer(commandBuffers[currentFrame], 0);
+
+  recordCommandBuffer(commandBuffers[currentFrame], imageIndex);
+
+  VkSubmitInfo submitInfo{};
+  submitInfo.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+
+  VkSemaphore waitSemaphores[] = {imageAvailableSemaphores[currentFrame]};
+  VkPipelineStageFlags waitStages[] = {
+      VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT};
+  submitInfo.waitSemaphoreCount = 1;
+  submitInfo.pWaitSemaphores = waitSemaphores;
+  submitInfo.pWaitDstStageMask = waitStages;
+  submitInfo.commandBufferCount = 1;
+  submitInfo.pCommandBuffers = &commandBuffers[currentFrame];
+  VkSemaphore signalSemaphores[] = {renderFinishedSemaphores[currentFrame]};
+  submitInfo.signalSemaphoreCount = 1;
+  submitInfo.pSignalSemaphores = signalSemaphores;
+
+  if (vkQueueSubmit(graphicsQueue, 1, &submitInfo,
+                    inFlightFences[currentFrame]) != VK_SUCCESS) {
+    throw std::runtime_error("failed to submit draw command buffer!");
+  }
+
+  VkPresentInfoKHR presentInfo{};
+  presentInfo.sType = VK_STRUCTURE_TYPE_PRESENT_INFO_KHR;
+
+  presentInfo.waitSemaphoreCount = 1;
+  presentInfo.pWaitSemaphores = signalSemaphores;
+
+  VkSwapchainKHR swapChains[] = {swapChain};
+  presentInfo.swapchainCount = 1;
+  presentInfo.pSwapchains = swapChains;
+  presentInfo.pImageIndices = &imageIndex;
+  presentInfo.pResults = nullptr;  
+
+  result = vkQueuePresentKHR(presentQueue, &presentInfo);
+  if (result == VK_SUBOPTIMAL_KHR) {
+    orientationChanged = true;
+  } else if (result == VK_ERROR_OUT_OF_DATE_KHR) {
+    recreateSwapChain();
+  } else if (result != VK_SUCCESS) {
+    throw std::runtime_error("failed to present swap chain image!");
+  }
+  currentFrame = (currentFrame + 1) % MAX_FRAMES_IN_FLIGHT;
+}
+
+void getPrerotationMatrix(const VkSurfaceCapabilitiesKHR &capabilities,
+                          const VkSurfaceTransformFlagBitsKHR &pretransformFlag,
+                          std::array<float, 16> &mat) {
+  // Identity Matrix
```

Comment:
so we don't need to handle 180 degree case right?

Reply:
nope!

Reply:
then we need to put a comment for that